src/EventStore.Core/TransactionLog/Chunks/TFChunk/TFChunk.cs

```diff
@@ -930,10 +930,11 @@ public enum ChunkVersions : byte {
 			writerWorkItem.Dispose();
 		}
 
-		public void Dispose() {
+		public void Dispose() => TryClose();
+
+		public bool TryClose() {
 			_selfdestructin54321 = true;
-			TryDestructFileStreams();
-			TryDestructMemStreams();
+			return TryDestructFileStreams() && TryDestructMemStreams();
```

Comment:
Intentional short circuit?

Reply:
:man_facepalming: well spotted, thanks will fix it!

Reply:
fixed: 5ede56ca6e117528881b36152a118b344432d62d